templates/config/rbac/leader_election_role.yaml.tpl

```diff
@@ -0,0 +1,37 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{.ServicePackageName}}-leader-election-role
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - configmaps
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
```

Comment:
I have confirmed that ACK does not need the `configmap`. This is sort of an either/or thing, where `configmap` or `leases` can be used, and the operator developer can choose which to use. For `legacy` purposes and for operators that might have been developed before `leases` were created, this is kept in the scaffolding.

Since ACK isn't exposing this [flag](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.16.0/pkg/manager/manager.go#L144-L175) which allows the choice, we are safe to remove the `confimap` section from the RBAC. 

@a-hilaly I hope all the above makes sense, let me know if there are questions...After this is removed I think we are good to go.